### DISCUSSION TITLE
How to select only those rows for which I know the uuid (and block select *) using RLS?

### DISCUSSION THREAD

**Discussion by ishanray** [supabase/supabase#7719] [Questions]
#### Table
id (uuid) | name (text)
------------------------

#### Policy
```
create policy "Allow select individual pad id" on pads
for select
using (
  position(id::text in current_query()) > 0
);
```

I have this policy which checks if the id is in the current query and then allow the select.

But this returns an empty `[]` set everytime. Am I missing something here?

**Comment by GaryAustin1** [comment]
Not sure current_query contains what you want:
For:
```
let response = await supabase
        .from('test4')
        .select('*')
        .eq('id2','0b92f77a-bcd7-4e70-8610-7331aa4f012d')
```
This is the query string:
```
query string = WITH pgrst_source AS ( SELECT "public"."test4".*FROM "public"."test4"  WHERE  "public"."test4"."id2" = $1  LIMIT $2 OFFSET $3 )  SELECT null::bigint AS total_result_set, pg_catalog.count(_postgrest_t) AS page_total, array[]::text[] AS header, coalesce(json_agg(_postgrest_t), '[]')::character varying AS body, nullif(current_setting('response.headers', true), '') AS response_headers, nullif(current_setting('response.status', true), '') AS response_status FROM ( SELECT * FROM pgrst_source ) _postgrest_t
```
The actual id is not in the string as it is a prepared statement...

I used this function as the policy to debug (it returns boolean)
```
begin
raise log 'query string = %',current_query();
return true;
end
```

**Reply by GaryAustin1** [reply]
A way to implement what you want (assuming there is no way to the parameters for the query) is to lock down table with select policy false and then use an rpc call with a security definer function to read the row with what ever checks you need.

**Reply by ishanray** [reply]
That is what I am using currently. I was just trying to move the database functions into policies, so that the business logic is all in one place, organized by tables. Your explanation makes sense though. I wish the query params could be exposed similarly as `request.jwt.claims` but I guess Postgrest does that?

**Reply by GaryAustin1** [reply]
PostgREST does not document being able to get the query or parameters.  I've not read thru the source code to see if there is something else, but PostgREST docs are normally pretty complete.

**Reply by steve-chavez** [reply]
> I wish the query params could be exposed similarly as request.jwt.claims but I guess Postgrest does that?

Not right now, but obtaining the request query parameters(with `request.params`) is planned: https://github.com/PostgREST/postgrest/pull/1710. There are other pressing issues right now, but I'll revisit that one after the upcoming launch week.

**Reply by ishanray** [reply]
That would be super useful. In supabase `auth.uid()` is really helpful and other functions like that would make it easy to write these kind of policies to restrict `select *`.

**Comment by EskelCz** [comment]
@steve-chavez How did it pan out with the query params? I need the same thing as the OP, but I also need it to work with realtime (where current_query doesn't work). For realtime RPC doesn't help me either, so I'm kinda stuck on a pretty basic thing. :(
